include/nighthawk/common/sink.h

```diff
@@ -0,0 +1,46 @@
+
+#pragma once
+
+#include <vector>
+
+#include "envoy/common/pure.h"
+
+#include "external/envoy/source/common/common/statusor.h"
+
+#include "api/client/service.grpc.pb.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Nighthawk {
+
+/**
+ * Abstract Sink interface.
+ */
+class Sink {
+public:
+  virtual ~Sink() = default;
+
+  /**
+   * Store an ExecutionResponse instance. Can be called multiple times for the same execution_id to
+   * persist multiple fragments that together will represent results belonging to a single
```

Comment:
Don't we need to somehow indicate when the last fragment was added? If we don't how will the system know if it got all the fragment or there was a crash in the middle of the execution?

Or is it that we don't know if we are getting all the fragments at this layer by design?

Reply:
That is right, the sink doesn't try to understand what values it is storing for a particular key.
Consumers of sink lookup data will be responsible for interpreting it.

Reply:
Thank you for explaining.